lib/active_shipping/location.rb

```diff
@@ -73,10 +73,11 @@ def self.from(object, options = {})
 
       ATTRIBUTE_ALIASES.each do |attribute, aliases|
         aliases.detect do |sym|
-          value = if hash_access
-            object[sym]
-          elsif object.respond_to?(sym)
-            object.send(sym)
+          value = object[sym] if hash_access
+          if !value &&
```

Comment:
Unfortunately complicated, do to how loosely we alloy data to be passed into this class.